### DISCUSSION TITLE
Vision OS support

### DISCUSSION THREAD

**Discussion by sandrohanea** [dotnet/maui#15539] [Ideas]
### Description

Will there be any support for Apple Vision Pro and Vision OS?

Probably to early to say how many changes are needed but intention of support is what I am looking for.

### Public API Changes

1. Support the platform as simple windows.
2. Add window depth (or Z) property 
3. Add windows MinimumDepth and MaximumDepth (if platform will support it) to be equivalent with MinimumWidth and MinimumHeight.
4. Add 3d object utility: create, move, allow interaction, etc. (maybe as 3rd party)

### Intended Use-Case

Platform will most likely evolve fast and will be a great point if Maui support will be added soon so Maui Adoption will also be driven by Vision adoption.

If people want to create new apps for this platform, Maui can be one of the initial options, depending on the business needs.

Also, existing Maui applications will have the ability to target this new platform, showing that Maui has the intention of remaining the “multi-platform app UI” even in the new horizons.

**Comment by mobilewares** [comment]
Seems like the Vision Pros are still a very long way off from public release - no initial release until early next year and that will be US only. So unless you have some advance access to these I wouldn't worry too much about it yet.
I imagine there will be some way to hack in support via platform specific code (in same way Google Auto/Carplay support is being done atm) even if there's no official support in MAUI.

**Reply by sandrohanea** [reply]
Simulator and SDK should be available soon (later this month), but indeed, they are not available yet and we could see the SDK capabilities just in some small presentations until now.

**Comment by drasticactions** [comment]
I moved this to Discussions. Considering that VisionOS would have to be added as a target to the underlying .NET iOS/Mac SDK and all that entails, support for this would be a very long time away and isn't worth tracking as an issue yet, IMO.

It's worth talking about what an API would look like, but in the end, it's too soon for this to be a mainline proposal.

**Comment by sandrohanea** [comment]
Thanks for the clarifications and conversion @drasticactions ,
I agree with you that it's too soon for this (we don't even know what API will support yet on the platform).

This month SDK and simulator should be available, so we can revise it a bit.

**Comment by guygodin** [comment]
SDK for visionOS has been released today in Xcode 15 beta 2. Hope to see this new platform supported!
https://developer.apple.com/download/all/?q=xcode%2015

**Comment by mhrastegari** [comment]
Meanwhile React Native:
![image](https://github.com/dotnet/maui/assets/42671084/02e1cbe3-28f5-4fe8-b7fa-c1a6ad79a4d9)

And Flutter:
![image](https://github.com/dotnet/maui/assets/42671084/cdb4d4ec-7a4b-451b-9ebd-1e25d8cebd7c)

**Reply by drasticactions** [reply]
Cool!

**Reply by Happypig375** [reply]
In .NET land we also have Avalonia btw
https://twitter.com/AvaloniaUI/status/1672289888937361413

**Reply by mhrastegari** [reply]
I wished I knew how they did it :)))))

**Comment by Xyncgas** [comment]
Maui.Blazor works

**Reply by mhrastegari** [reply]
HOW??? I was trying to do it all night!

**Reply by mhrastegari** [reply]
Is the command anything more than this?

`dotnet build -t:Run /p:_DeviceName=:v2:udid=APPLE_VISION_UDID -f net7.0-ios`

**Reply by bonniemathew** [reply]
I am also curious to know if there is a way to run the existing dotnet iOS App on Vision OS.

**Comment by akhanalcs** [comment]
This would be very cool!

**Reply by amoffatt** [reply]
Any updates on this now that the Vision Pro has been out in the wild for awhile?
